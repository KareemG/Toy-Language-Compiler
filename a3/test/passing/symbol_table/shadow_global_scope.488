{
    function test : integer
    {
        var a : integer

        a := 5

        return with a
    }

    var a : integer
    var b : integer

    a := 5

    b := test

    if b = 5 then
        write "valid"
    else
        write "invalid"
}