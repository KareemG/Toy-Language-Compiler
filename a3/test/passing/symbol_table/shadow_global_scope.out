{
 Symbols:
 b: integer
 a: integer
 test: integer
 test : integer{
 Symbols:
 a: integer
 }